src/pkg/caendr/caendr/models/datastore/dataset_release.py

```diff
@@ -217,7 +217,7 @@ def check_fasta_file_exists(self):
   @staticmethod
   def get_fasta_filepath_url_template():
     obj = DatasetRelease.get_fasta_filepath_obj_template()
-    return TokenizedString( generate_blob_url(obj['bucket'], f'{ obj["path"] }/{ obj["name"] }{ obj["ext"] }') )
+    return TokenizedString.apply( generate_blob_uri, obj['bucket'], obj['path'], obj['name'] + obj['ext'], schema=BlobURISchema.HTTPS )
```

Comment:
The `apply` method applies a given function (`generate_blob_uri` here) to a set of `TokenizedString` arguments, handling the tokenization implicitly.

Side note: I originally designed the `TokenizedString` class to be able to carry tokens around, but now that I've been using it for a bit, I've found that there's almost never a practical use case for that (you pretty much always know all the tokens you want to use at the same time).  I don't want to focus too much energy on it now, since it does work, but I think a more streamlined class that just validates token replacements would be (a) equally useful and (b) less awkward.